### PR TITLE
feat(components): Add gap prop into Flex component and Grid component

### LINKED ISSUE #800: Add `gap` prop to Flex / Grid
Expose the [gap](https://developer.mozilla.org/en-US/docs/Web/CSS/gap) prop to Flex / Grid, including `row-gap` and `column-gap` as well.

[Browser compatibility](https://developer.mozilla.org/en-US/docs/Web/CSS/gap#browser_compatibility) seems to be good to go.

### PR DESCRIPTION
**Pull request by MariaJose**
## What?
Expose the [gap](https://developer.mozilla.org/en-US/docs/Web/CSS/gap) prop to Flex / Grid, including row-gap and column-gap. Also, update the corresponding documentation pages to let the users know about this new available props.

Props for Flex component:
- flexGap
- flexRowGap
- flexColumnGap

Props for Grid component:
- gridGap
- gridRowGap
- gridColumnGap

## Why?
This was a reported enhancement(Fixes #800). These new props are gonna help the user to easily apply spacing between items. 

## Screenshots/Screen Recordings
<table>
<tr>
	<td>Grid component
	<td>Flex component
<tr>
<tr>
       <td><video src= "https://user-images.githubusercontent.com/39739180/171470343-79fa321a-caf9-4fc0-8950-f2476f360a16.mp4"/>
	<td><video src= "https://user-images.githubusercontent.com/39739180/171470374-a8c24154-798c-49a4-82dd-87cd60d107ea.mp4" />
<tr>
<tr>
       <td><video src="https://user-images.githubusercontent.com/39739180/171470845-9da43b1c-1c70-4ccb-8f69-993a43f50c75.mp4" />
	<td><video src="https://user-images.githubusercontent.com/39739180/171470867-9117be02-62b9-4e32-94ee-d1096f686704.mp4" />
<tr>
<tr>
	<td><video src="https://user-images.githubusercontent.com/39739180/171471039-108a2fec-2c36-4af4-a5d4-77ff3695ed56.mp4" />
	<td><video src="https://user-images.githubusercontent.com/39739180/171471104-d6535475-264a-4b4c-86ed-32b13ba7e157.mp4" />
<tr>
</table>

## Testing/Proof
Run tests
Test locally

### CODE REVIEW

File: packages/big-design/src/components/Alert/__snapshots__/spec.tsx.snap [hunk 1]
```diff
@@ -330,7 +330,7 @@ exports[`render warning Alert 1`] = `
 }
 
 .c1 {
-  grid-gap: 1rem;
+  gap: 1rem;
   display: grid;
```
Comment (MariaJose) [inline packages/big-design/src/components/Alert/__snapshots__/spec.tsx.snap:334]: not a biggy, but noticed that some components use the `grid-gap` internally, should we update it to the new name `gap` css prop? If so, this change should be made in another PR?

Example:
```jsx
export const StyledInlineMessage = styled(Grid)<InlineMessageProps>`
  border: ${({ theme }) => theme.border.box};
  border-radius: ${({ theme }) => theme.borderRadius.normal};
  grid-gap: ${({ theme }) => theme.spacing.xSmall};
  grid-template-columns: ${({ theme }) => `${theme.spacing.large} 1fr ${theme.spacing.large}`};
  padding: ${({ theme }) => theme.spacing.xSmall};
`
``` 

```jsx
export const StyledInlineMessage = styled(Grid)<InlineMessageProps>`
  border: ${({ theme }) => theme.border.box};
  border-radius: ${({ theme }) => theme.borderRadius.normal};
  gap: ${({ theme }) => theme.spacing.xSmall};
  grid-template-columns: ${({ theme }) => `${theme.spacing.large} 1fr ${theme.spacing.large}`};
  padding: ${({ theme }) => theme.spacing.xSmall};
`
```

**Comment by bc-tgomez** [comment]
Hey there @MariaJose 👋  @bc-andreadao is out this week on planned vacation. Do you require this review ASAP or would it be ok for her to handle the edits when she gets back?

**Comment by MariaJose** [comment]
> Hey there @MariaJose 👋 @bc-andreadao is out this week on planned vacation. Do you require this review ASAP or would it be ok for her to handle the edits when she gets back?

 Hey @bc-tgomez 👋🏼  thanks for letting me know! I think is not an urgency, so we can wait for her next week or feel free to check the edits when having a chance. Either way could work :)

File: packages/docs/PropTables/FlexPropTable.tsx [hunk 2]
```diff
@@ -35,6 +45,24 @@ const flexProps: Prop[] = [
       </>
     ),
   },
+  {
+    name: 'flexGap',
+    types: 'string',
+    description: (
+      <>
+        Controls the spacing between flex items. Same as the <Code highlight={false}>gap (grid-gap)</Code> CSS property.
```
[the remark this answers is not in the document]
Comment (MariaJose) [reply packages/docs/PropTables/FlexPropTable.tsx:53]: good point 👍🏼

File: packages/big-design/src/components/Flex/withFlex.ts [hunk 3]
```diff
@@ -6,7 +6,10 @@ import { FlexedItemProps, FlexedOverload, FlexedProps } from './types';
 export const withFlexedContainer = () => css<FlexedProps>`
   ${({ alignContent, theme }) => alignContent && getFlexedStyles(alignContent, theme, 'align-content')};
   ${({ alignItems, theme }) => alignItems && getFlexedStyles(alignItems, theme, 'align-items')};
+  ${({ flexColumnGap, theme }) => flexColumnGap && getFlexedStyles(flexColumnGap, theme, 'column-gap')};
   ${({ flexDirection, theme }) => flexDirection && getFlexedStyles(flexDirection, theme, 'flex-direction')};
+  ${({ flexGap, theme }) => flexGap && getFlexedStyles(flexGap, theme, 'gap')};
```
Comment (jorgemoya) [inline packages/big-design/src/components/Flex/withFlex.ts:11]: What happens if we have `gap`, `column-gap`, & `row-gap` props in the same element? Does `gap` supersede the others or does it still depends on the cascade? Wondering if we should move `gap` before both `column-gap` and `row-gap`.

Comment (MariaJose) [reply packages/big-design/src/components/Flex/withFlex.ts:11]: ohh yeah good catch, thanks!  I think it's gonna overrides. I agree, the gap should be before column and row 👍🏼

**Review by jorgemoya — APPROVED** [review]
lgtm!

File: packages/big-design/src/components/Flex/types.ts [hunk 4]
```diff
@@ -25,8 +25,22 @@ type AlignSelf = ResponsiveProp<
   'auto' | 'normal' | 'self-start' | 'self-end' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch'
 >;
 
+type FlexBasis = ResponsiveProp<'auto' | 'fill' | 'min-content' | 'max-content' | 'fit-content' | 'content' | string>;
+
+type FlexColumnGap = ResponsiveProp<string>;
+
 type FlexDirection = ResponsiveProp<'row' | 'column' | 'row-reverse' | 'column-reverse'>;
 
+type FlexGap = ResponsiveProp<string>;
```
Comment (icatalina) [inline packages/big-design/src/components/Flex/types.ts:34]: 🤔 why aren't we using a `Spacing` prop for this?

Comment (MariaJose) [reply packages/big-design/src/components/Flex/types.ts:34]: Good point, I think I added it because I was following the css api, but makes sense that we should follow BD spacing values. I'll create an issue later today about this. Thanks for the feedback!!
